src/dmd/root/filename.d

```diff
@@ -1078,10 +1125,10 @@ version (Posix)
     */
     auto absPathThen(alias F)(const(char)[] fileName)
     {
-        import core.sys.posix.stdlib: realpath, free;
-        char* absPath = fileName.toCStringThen!((fn) => realpath(&fn[0], null /* realpath allocates */));
-        scope(exit) free(absPath);
-        return F(absPath.toDString());
+        import core.sys.posix.stdlib: free;
+        auto absPath = FileName.canonicalName(fileName);
+        scope(exit) free(cast(void*)absPath.ptr);
+        return F(cast(char[])absPath);
```

Comment:
It would be simpler just to swap lines 1130 and 1131:
```
F(..)
free(...)
```

Reply:
If this was a void function, then I'd agree.